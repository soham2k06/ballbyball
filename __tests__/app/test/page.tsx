import TestPage from "@/app/test/page";
import { fireEvent, render, screen } from "@testing-library/react";

// test("Testing Input box", () => {
//   render(<TestPage />);
//   const input = screen.getByRole("textbox") as HTMLInputElement;

//   expect(input).toBeInTheDocument();
//   fireEvent.change(input, { target: { value: "Hello" } });
//   expect(input.value).toBe("Hello");
// });

// test("Testing Dark mode button click", () => {
//   render(<TestPage />);
//   const button = screen.getByRole("button") as HTMLButtonElement;
//   const mode = screen.getByText(/Mode:/i);

//   // check if button and mode are present
//   expect(button).toBeInTheDocument();

//   // click the button, so the mode should change
//   fireEvent.click(button);

//   // check if the mode is changed, because it was light before
//   expect(mode).toHaveTextContent("dark");
// });

test("Testing Test page", () => {
  const page = render(<TestPage />);

  expect(page).toMatchSnapshot();
});
