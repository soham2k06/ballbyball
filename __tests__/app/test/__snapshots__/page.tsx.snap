// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Test page 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mx-auto max-w-7xl p-8"
      >
        <h1
          class="mb-8 text-2xl font-semibold"
        >
          Test cases
        </h1>
        <input
          id="name"
          placeholder="Enter..."
          type="text"
          value=""
        />
        <button>
          Update Mode
        </button>
        <p>
          Mode: 
          light
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mx-auto max-w-7xl p-8"
    >
      <h1
        class="mb-8 text-2xl font-semibold"
      >
        Test cases
      </h1>
      <input
        id="name"
        placeholder="Enter..."
        type="text"
        value=""
      />
      <button>
        Update Mode
      </button>
      <p>
        Mode: 
        light
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
